### ISSUE TITLE
Maestro promotion pipeline fails when publishing to multiple channel in VS-test repo 

### ISSUE THREAD

**Issue by epananth** [dotnet/arcade#13176]
- [ ] This issue is blocking 
- [ ] This issue is causing unreasonable pain

| Pipeline | Channel | 
|---|---|
| https://dev.azure.com/dnceng/internal/_build/results?buildId=2159186&view=results | .NET Core Tooling Dev', '.NET 7', '.NET 8| 
| https://dev.azure.com/dnceng/internal/_build/results?buildId=2151357&view=results | .NET Core Tooling Dev', '.NET 7', '.NET 8| 

NOTE : 
1) The build fails on the first run 
2) On retrying the build, the build is successful 
3) All the channels have different feeds 

### Release Note Category
- [ ] Feature changes/additions 
- [ ] Bug fixes
- [ ] Internal Infrastructure Improvements

**Comment by epananth** [comment]
@MichaelSimons Vs-test should only publish to VS channels. Any idea why it is publishing to other channels? 

Not sure you are the right contact for this, can you please help?

**Comment by MichaelSimons** [comment]
> Not sure you are the right contact for this, can you please help?

I am not the right contact - I would ask @nohwnd, @Evangelink, or @mmitche

**Comment by nohwnd** [comment]
We want to insert into both net7 sdk (or in broader terms to the the currently developed stable version), and net8 (to the next version), we are also pushing into .net tooling because .net runtime is using us for testing, @ViktorHofer  please confirm this :) 

If there is a better configuration to have setup please let us know. 

Also how do I see where is VS17.7 pushing?

**Comment by mmitche** [comment]
For tooling repos, pushing into the VS channels is the right way to make it into the SDK, since there are multiple bands of the SDK that go into different versions of VS. Pushing to the .NET 7, .NET 6, etc. channels is not recommended. Runtime does not have a dependency on vstest. They probably use you via the SDK. The only subscription that does source from .NET Core Tooling Dev is vs-code-coverage. https://dev.azure.com/devdiv/DevDiv/_git/vs-code-coverage. Since vs-code-coverage also has VS-centric branches (rel/17.N), I suspect that it should actually be sourcing from "VS 17.N" channels.

For instance, msbuild will push a branch to VS 17.6, which will go into the 7.0.3xx SDK, which goes into VS 17.6. They also have a branch for 17.4, which goes into 7.0.1xx. There are some cases when these overlaps. VS 17.7, for instance, goes into **both** 7.0.4xx (which will go into VS 17.7) as well as .NET 8.0.1xx (for .NET previews).

Basically, if you're a tooling repo, and you have a 1:many relationship with the .NET SDKs, the correct thing is to push into the VS channels. This is what is currently set up for vstest. There should probably be a subscription to 8.0.1xx from 17.7 too.

```
PS C:\r\windowsdesktop> darc get-subscriptions --source-repo vstest --target-repo sdk --channel "VS 17.7"
https://github.com/microsoft/vstest (VS 17.7) ==> 'https://github.com/dotnet/sdk' ('release/7.0.4xx')
  - Id: f6e9d7f8-05b3-4da3-2521-08db309217d3
  - Update Frequency: EveryBuild
  - Enabled: True
  - Batchable: False
  - PR Failure Notification tags:
  - Merge Policies:
    Standard
  - Last Build: 20230602.1 (1fe03b9451e6605b0846fc544cc0effd2b39a85c)
```

/cc @marcpopMSFT

**Comment by nohwnd** [comment]
I've updated the subscriptions, we now use VS17.7 as our default channel on main, and subscribe to that from dotnet/sdk main, release/7.0.4xx and code-coverage.

**Comment by nohwnd** [comment]
@epananth do you want to double check something or should I close?

**Comment by epananth** [comment]
I checked the latest build -> https://dev.azure.com/dnceng/internal/_build/results?buildId=2195441&view=results

its working.. Hurraaay!!

Closing this issue :)
